### ISSUE TITLE
Ability for sliders to have log scale

### ISSUE THREAD

**Issue by cbanek** [bokeh/bokeh#8012]
This is a feature request, as I haven't determined how I can do this yet.

I have a bokeh graph, with a y axis that is log scale, ans x axis that is linear/normal scale.  I have two RangeSliders that allow for filtering on the graph based on the values of the sliders.  The RangeSliders are always linear, so it can be hard to select smaller values, since step and the scale are all linear.  Is there any way to be able to specify a scale, kind of like how axis is done, for a slider, that would allow the value of the slider to match with a log scale axis?

**Comment by mattpap** [comment]
No, currently only linear sliders are supported. The underling library that we use, noUiSlider, allows for log scales, so this could be implemented, though I can't give you any guarantees when, if at all. However, there is a new feature called range tool, that will be part of 0.13.0 release (tomorrow), that could be used instead or maybe is even a better choice altogether. [Here](http://bokeh.pydata.org/en/dev/docs/gallery/range_tool.html)'s an example (only one linear axis, but it can be extended support your use case).

**Comment by cbanek** [comment]
That's excellent, I was hoping I missed something like that.  That example looks super sweet.  No rush on this either.  Thanks for the input.

**Comment by chrisrb10** [comment]
Hi - was the (potential) feature to have a log scale on RangeSlider ever implemented? I have a use case that needs exactly that.

**Comment by bryevdv** [comment]
If any progress had been made it would have been recorded here.

**Comment by randompast** [comment]
In the meantime, maybe you could make a slider l,r = [3,7] and update your parameter with v = 10**-r.

**Comment by Hoeijmakers** [comment]
I was searching for this feature today, and I would like to reiterate this feature request. The proposed solution to map the variable to its logarithm works, but is inconvenient as it requires the user to mentally take the exponent of the labels shown on the slide bar.

**Comment by xavArtley** [comment]
For the moment you can change the displayed value using a Tickformatter:
```python
s = Slider(start=0, end=1, value=0.5, step=0.1, format=FuncTickFormatter(code="return Math.exp(tick).toFixed(2)"))
```

**Comment by Hoeijmakers** [comment]
> For the moment you can change the displayed value using a Tickformatter:
> 
> ```python
> s = Slider(start=0, end=1, value=0.5, step=0.1, format=FuncTickFormatter(code="return Math.exp(tick).toFixed(2)"))
> ```

Dear xavArtley; thanks for that lightning-fast suggestion. I confirm that it is working neatly (actually applied to a RangeSlider).

**Comment by phurwicz** [comment]
> For the moment you can change the displayed value using a Tickformatter:
> 
> ```python
> s = Slider(start=0, end=1, value=0.5, step=0.1, format=FuncTickFormatter(code="return Math.exp(tick).toFixed(2)"))
> ```

Wonderful! Being able to format the tick is all it really takes.
